nutaudet/HPT.cxx

```diff
@@ -232,12 +232,9 @@ void Hpt::ConstructGeometry()
     if (fDesign == 3){
         //Trackers that in design 3 follow the target --------------------------------------------------------------------------------------
         TGeoVolume *volMagRegion=gGeoManager->GetVolume("volMagRegion");
-        TGeoVolume *volTarget =gGeoManager->GetVolume("volTarget");
-        TGeoVolume *tTauNuDet = gGeoManager->GetVolume("tTauNuDet");
+        TGeoVolume *volTarget =gGeoManager->GetVolume("volTarget");        
 
         Double_t DZMagnetizedRegion = ((TGeoBBox*) volMagRegion->GetShape())->GetDZ() *2;
-        Double_t DYMagnetizedRegion = ((TGeoBBox*) volMagRegion->GetShape())->GetDY() *2;
-        Double_t DXMagnetizedRegion = ((TGeoBBox*) volMagRegion->GetShape())->GetDX() *2;
 
         Double_t DZTarget = ((TGeoBBox*) volTarget->GetShape())->GetDZ() *2;
```

Comment:
Could you fix this C-style cast please?

Reply:
Sorry, I did not get this comment. 

The cast is required in order to access the GetDX() method being required from the TGeoBBox class, while TGeoVolume::GetShape() returns a generic TGeoShape object:

https://root.cern/doc/master/classTGeoVolume.html#aaf5dc92206d7bc56e7bdb9f84287c2fe

Reply:
C-style casts can easily lead to issues. A cast here is needed, but should be a `static_cast`, which is stricter and easier to search for.

See e.g.:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Reply:
See also https://github.com/ShipSoft/FairShip/wiki/CPP-guidelines

Reply:
I confirm that after the rebase the compilation and test simulation worked succesfully. 
I am now going to force push to apply it.